docsite/rst/developing_program_flow_modules.rst

```diff
@@ -0,0 +1,392 @@
+  - :code:`from ansible.module_utils.MOD_LIB_NAME import *` is replaced with the
+    contents of the :file:`ansible/module_utils/MOD_LIB_NAME.py`  These should
+    only be used with :ref:`new-style python modules <flow_python_modules>`.
+  - :code:`#<<INCLUDE_ANSIBLE_MODULE_COMMON>>` is equivalent to
+    :code:`from ansible.module_utils.basic import *` and should also only apply
+    to new-style python modules.
+  - :code:`# POWERSHELL_COMMON` substitutes the contents of
+    :file:`ansible/module_utils/powershell.ps1`.  It should only be used with
+    :ref:`new-style Powershell modules <flow_powershell_modules>`.
+
+* Replacements that are used by ansible ``module_utils`` code.  These are internal
+  replacement patterns.  They may be used internally in the above public
+  replacements but shouldn't be used directly by modules.
+
+  - :code:`"<<ANSIBLE_VERSION>>"` is substituted with the ansible version.  In
+    a new-style python module, it's better to use ``from ansible import
+    __version__`` and then use ``__version__`` instead.
+  - :code:`"<<INCLUDE_ANSIBLE_MODULE_COMPLEX_ARGS>>"` is substituted with
+    a string which is the python ``repr`` of the :term:`JSON` encoded module
+    parameters.  Using repr on the JSON string makes it safe to embed in
+    a python file.  In :ref:`new-style python modules <flow_python_modules>`
+    under :ref:`ziploader` this will be passed in via an environment variable
+    instead.
+  - :code:`<<SELINUX_SPECIAL_FILESYSTEMS>>` substitutes a string which is
+    a comma separated list of filesystems which have a file system dependent
+    security context in selinux.  In new-style python modules, this will be
+    found by looking up ``SELINUX_SPECIAL_FS`` from the
+    :envvar:`ANSIBLE_MODULE_CONSTANTS` environment variable.  See the
+    :ref:`ziploader` documentation for details.
+  - :code:`<<INCLUDE_ANSIBLE_MODULE_JSON_ARGS>>` substitutes the module
+    parameters as a json string.  Care must be taken to properly quote the
+    string as JSON data may contain quotes.  JSON_ARGS is not substituted in
+    new-style python modules as they can get the module parameters via the
+    environment variable.
+  - the string :code:`syslog.LOG_USER` is replaced wherever it occurs with the
+    value of ``syslog_facility`` from the :file:`ansible.cfg` or any
+    ``ansible_syslog_facility`` inventory variable that applies to this host.  In
+    new-style python modules you can get the value of the ``syslog_facility``
+    by looking up ``SYSLOG_FACILITY`` in the :envvar:`ANSIBLE_MODULE_CONSTANTS`
+    environment variable.  See the :ref:`ziploader` documentation for details.
+
+.. _ziploader:
+
+ziploader
+^^^^^^^^^
+
+Ziploader differs from :ref:`module_replacer` in that it uses real python
+imports of things in module_utils instead of merely preprocessing the module.
+It does this by constructing a zipfile of the module, files in
+:file:`ansible/module_utils` that are imported by the module, and some
```

Comment:
I see what you mean changed to this:

It does this by constructing a zipfile--which includes the module file, files in :file:ansible/module_utils that are imported by the module, and some boilerplate content to pass in the constants.